pros/src/adi/analog.rs

```diff
@@ -0,0 +1,116 @@
+use crate::adi::{
+    AdiError,
+    AdiSlot
+};
+
+use crate::error::bail_on;
+
+use pros_sys::PROS_ERR;
+
+pub struct AdiAnalogIn {
+    port: u8,
+}
+
+impl AdiAnalogIn {
+    /// Create an AdiAnalogIn without checking if it is valid.
+    /// 
+    /// # Safety
+    /// 
+    /// The port must be above 0 and below [`pros_sys::NUM_ADI_PORTS`].
+    pub unsafe fn new_unchecked(slot: AdiSlot) -> Self {
+        Self {
+            port: slot as u8
+        }
+    }
+    
+    /// Create an AdiAnalogIn, panicking if the port is invalid.
+    pub fn new_raw(slot: AdiSlot) -> Self {
+        let port = slot as u8;
+        if port < 1 || port > {pros_sys::NUM_ADI_PORTS as u8} {
+            panic!("Invalid ADI port");
+        }
+        Self { port }
+    }
+
+    /// Create an AdiAnalogIn, returning err `AdiError::InvalidPort` if the port is invalid.
+    pub fn new(slot: AdiSlot) -> Result<Self, AdiError> {
+        let port = slot as u8;
+        if port < 1 || port > {pros_sys::NUM_ADI_PORTS as u8} {
+            return Err(AdiError::InvalidPort);
+        }
+        Ok(Self { port })
+    }
+
+    /// Calibrates the analog sensor on the specified channel.
+    ///
+    /// This method assumes that the true sensor value is not actively changing at this time and computes an average from approximately 500 samples, 1 ms apart, for a 0.5 s period of calibration. The average value thus calculated is returned and stored for later calls to the adi_analog_read_calibrated and adi_analog_read_calibrated_HR functions. These functions will return the difference between this value and the current sensor value when called.
```

Comment:
This should be split into multiple lines. Really long lines in doc comments are much harder to read in code.